core/jazz_admin/index.js

```diff
@@ -25,30 +25,79 @@ const errorHandlerModule = require("./components/error-handler.js"); //Import th
 const responseObj = require("./components/response.js"); //Import the response module.
 const configObj = require("./components/config.js"); //Import the environment data.
 const logger = require("./components/logger.js"); //Import the logging module.
-const request = require('request');
-function handler(event, context, cb) {
+const crud = require("./components/crud")();
+
+const handler = (event, context, cb) => {
   //Initializations
-  var errorHandler = errorHandlerModule();
-  var config = configObj.getConfig(event, context);
+  let errorHandler = errorHandlerModule();
+  let config = configObj.getConfig(event, context);
+  global.config = config;
   logger.init(event, context);
 
   try {
-    var apiResponseObj = {};
+
+    if (event && !event.method) {
+      return cb(JSON.stringify(errorHandler.throwInputValidationError("Method cannot be empty")));
+    }
+
+    if (event && !event.principalId) {
+      return cb(JSON.stringify(errorHandler.throwUnauthorizedError("User is not authorized to access this service|Authorization Incomplete")));
+    }
+
+    if (event && event.principalId != config.ADMIN_ID) {
+      return cb(JSON.stringify(errorHandler.throwUnauthorizedError("This user is not authorized to access this service.")));
+    }
+
     if (event && event.method && event.method === 'GET') {
-      if (!event.principalId) {
-        logger.error('Authorizer did not send the user information, please check if authorizer is enabled and is functioning as expected!');
-        return cb(JSON.stringify(errorHandler.throwUnauthorizedError("User is not authorized to access this service|Authorization Incomplete")));
+      let apiResponseObj = {};
+      exportable.getConfiguration()
+        .then((res) => {
+          apiResponseObj.config = res;
+          return cb(null, responseObj(apiResponseObj, event.body));
+        }).catch((error) => {
+          logger.error("Failed to load admin configuration :" + JSON.stringify(error));
+          return cb(JSON.stringify(errorHandler.throwInternalServerError("Failed to load admin configuration.")));
+        });
+
+    } else if (event && event.method && event.method === 'POST') {
+      if (event && !event.body) {
```

Comment:
why not move all these validations(including lines 39-49) to a separate file. It will make the code look cleaner

Reply:
Now only validating the event.method. In future we need to validate the incoming fields, so will move the entire validation to the separate file as part of that.